src/TabContainer.js

```diff
@@ -94,9 +94,7 @@ const TabContainer = props => {
   } = useUncontrolled(props, { activeKey: 'onSelect' });
 
   const generateChildId = useMemo(
-    () =>
-      generateCustomChildId ||
-      ((key, type) => (id ? `${id}-${type}-${key}` : null)),
+    () => generateCustomChildId || ((key, type) => `${id}-${type}-${key}`),
```

Comment:
why remove the null fallback?

Reply:
Since at least one of id and generateChildId must be defined, id cannot be a falsy value if generateCustomChildId is falsy, as validated in propTypes. So the null fallback is unnecessary in my opinion.

Reply:
they're semantically required, yes, but the issue is that there are no hard checks preventing users from omitting one of them.

what we're looking for, then, is the least broken behavior possible in the event that users specify neither

Reply:
I can revert the refactor commit

Reply:
yes please; thanks!

Reply:
Done!